hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java

```diff
@@ -469,6 +469,25 @@ public OzoneOutputStream createKey(String key, long size,
         .createKey(volumeName, name, key, size, replicationConfig, keyMetadata);
   }
 
+  /**
+   * Overwrite an existing key using optimistic locking. The existingKey must exist in Ozone to allow
+   * the new key to be created with the same name. Additionally, the existingKey must not have been
+   * modified since the time its details were read. This is controlled by the objectID and updateID
+   * fields in the existingKey. If the key is replaced the objectID will change. If it has been updated
+   * (eg appended) the updateID will change. If either of these have changed since the existingKey
+   * was read, either the initial key create will fail, or the key will fail to commit after the data
+   * has been written.
+   *
+   * @param existingKey       Name of the key to be created.
+   * @param replicationConfig Replication configuration.
+   * @return OzoneOutputStream to which the data has to be written.
+   * @throws IOException
+   */
+  public OzoneOutputStream overWriteKey(OzoneKeyDetails existingKey, ReplicationConfig replicationConfig)
```

Comment:
nit: "overwrite" is used as one word in comments and some variable names, so I think method name should match that.

```suggestion
  public OzoneOutputStream overwriteKey(OzoneKeyDetails existingKey, ReplicationConfig replicationConfig)
```